subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -158,6 +167,26 @@ private void checkProperties(Properties existingProperties) {
         }
     }
 
+    private static final String DISTRIBUTION_URL_EXCEPTION_MESSAGE = "Test of distribution url %s failed. Please check the values set with --gradle-distribution-url and --gradle-version.";
+
+    private void testDistributionUrl() {
+        if (distributionUrlConfigured) {
+            String url = getDistributionUrl();
+            URI uri = URI.create(url);
+            if (uri.getScheme().equals("file")) {
+                if (!Files.exists(Paths.get(uri).toAbsolutePath())) {
+                    throw new UncheckedIOException(String.format(DISTRIBUTION_URL_EXCEPTION_MESSAGE, url));
+                }
+            } else if (uri.getScheme().startsWith("http") && !isOffline) {
```

Comment:
💭 As discussed, let's create an issue that mandates `https` here unless you explicit opt-out

Reply:
https://github.com/gradle/gradle/issues/24367